packages/client/src/utils.ts

```diff
@@ -68,3 +76,39 @@ export const createDefaultOnRedirect = () => {
     window.location.assign(url);
   };
 };
+
+export const generateCallbackUri = (uriInfo: UriInfo): string => {
+  const callbackUriBase: string = uriInfo.redirectUri
+    .slice()
+    .concat(uriInfo.code || uriInfo.state || uriInfo.error || uriInfo.errorDescription ? '?' : '');
+
+  const callbackUriConcatCode: string = callbackUriBase.concat(
+    uriInfo.code ? `code=${uriInfo.code}` : ''
+  );
+
+  const callbackUriConcatState: string = callbackUriConcatCode.concat(
```

Comment:
we do have[ query-string](https://github.com/sindresorhus/query-string) added as a dependency. Could leverage the library to handle the search params and append it to the `redirectUri`.   

1. retrieve any original query params from the `redirectUri`
2. merge the query params with all the `non-empty` callback params
3. generate the new callbackUri